incubator/hnc/internal/validators/object.go

```diff
@@ -244,16 +246,19 @@ func (o *Object) handleInherited(op admissionv1beta1.Operation, newSource, oldSo
 	return deny(metav1.StatusReasonInternalError, "unknown operation: "+string(op))
 }
 
-// isDeletingNS returns true if the namespace of the object is already being deleted, based on the
-// in-memory forest.
-func (o *Object) isDeletingNS(inst *unstructured.Unstructured) bool {
-	o.Forest.Lock()
-	defer o.Forest.Unlock()
-	ns := o.Forest.Get(inst.GetNamespace())
-	if ns == nil {
+// isDeletingNS returns true if the namespace of the object is already being deleted
+func (o *Object) isDeletingNS(ctx context.Context, inst *unstructured.Unstructured) bool {
+
+	ns, err := o.K8sClient.CoreV1().Namespaces().Get(ctx, inst.GetNamespace(), metav1.GetOptions{})
```

Comment:
Using the existing client, this should be something like:

```
ns := &corev1.Namespace{}
nnm := metav1.NamespacedName{Name: inst.GetNamespace()}
err := o.client.Get(nnm, ns)
```

But I'm not 100% sure about all the package names and stuff.

It's probably also ok to skip this check if `o.client` is nil, which will be the case in most of the unit tests.

Reply:
Updated in latest commit